pantheon/src/main/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommand.java

```diff
@@ -0,0 +1,300 @@
+
+    /** Generates nodes keypairs. */
+    private void generateNodesKeys() {
+      final int nodesCount = nodesConfig.getInteger("count");
+      LOG.info("Generating {} nodes keys.", nodesCount);
+      IntStream.range(0, nodesCount).forEach(this::generateNodeKeypair);
+    }
+
+    /**
+     * Generate a keypair for a node.
+     *
+     * @param node The number of the node.
+     */
+    private void generateNodeKeypair(final int node) {
+      try {
+        LOG.info("Generating keypair for node {}.", node);
+        final SECP256K1.KeyPair keyPair = SECP256K1.KeyPair.generate();
+        writeKeypair(keyPair.getPublicKey(), keyPair.getPrivateKey());
+
+      } catch (IOException e) {
+        LOG.error("An error occurred while trying to generate node keypair.", e);
+      }
+    }
+
+    /**
+     * Writes public and private keys in separate files. Both are written in the same directory
+     * named with the address derived from the public key.
+     *
+     * @param publicKey The public key.
+     * @param privateKey The private key. No file is created if privateKey is NULL.
+     * @throws IOException
+     */
+    private void writeKeypair(
+        final SECP256K1.PublicKey publicKey, final SECP256K1.PrivateKey privateKey)
+        throws IOException {
+      final Address nodeAddress = Util.publicKeyToAddress(publicKey);
+      addressesForGenesisExtraData.add(nodeAddress.toUnprefixedString());
+      final Path nodeDirectoryPath = keysDirectory.resolve(nodeAddress.toString());
+      Files.createDirectory(nodeDirectoryPath);
+      createFileAndWrite(nodeDirectoryPath, publicKeyFileName, publicKey.toString());
+      if (privateKey != null) {
+        createFileAndWrite(nodeDirectoryPath, privateKeyFileName, privateKey.toString());
+      }
+    }
+
+    /**
+     * Computes RLP encoded exta data from pre filled list of addresses.
+     *
+     * @throws IOException
+     */
+    private void processExtraData() throws IOException {
+      LOG.info("Generating genesis extra data.");
+      final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+      final String jsonAddresses = gson.toJson(addressesForGenesisExtraData);
+      final String extraData =
+          RLPType.IBFT_EXTRA_DATA.getAdapter().encode(jsonAddresses).toString();
+      genesisConfig.put("extraData", extraData);
+    }
+
+    private void createFileAndWrite(
+        final Path directory, final String fileName, final String content) throws IOException {
+      final Path filePath = directory.resolve(fileName);
+      Files.createFile(filePath);
+      Files.write(filePath, content.getBytes(UTF_8));
+    }
+
+    /**
+     * Parses the root configuration file and related sub elements.
+     *
+     * @throws IOException
+     */
+    private void parseConfig() throws IOException {
+      final String configString =
+          Resources.toString(configurationFile.toPath().toUri().toURL(), UTF_8);
+      operatorConfig = new JsonObject(configString);
```

Comment:
Won't you prefer a Java object mapping instead of a generic gson object and manual key retrieval ? That would give a first pass validation of the JSON config provided and ease the use of the data. If this is done on purpose, please explain.

Reply:
In this case i prefer to do like this because i use only few JSON items. Moreover it is easier to split handling of specific JSON node in different methods, special cases with empty values are also easier to manage. Also don't want to create a test specific object when not necessarily required.